src/GraphQL/Language/CoreToVanillaConverter.cs

```diff
@@ -231,7 +231,18 @@ public IValue Value(GraphQLValue source)
                 case ASTNodeKind.FloatValue:
                 {
                     var str = (GraphQLScalarValue)source;
-                    return new FloatValue(ValueConverter.ConvertTo<double>(str.Value)).WithLocation(str, _body);
+
+                    // the idea is to see if there is a loss of accuracy of value
+                    // for example, 12.1 or 12.11 is double but 12.10 is decimal
+                    double dbl = ValueConverter.ConvertTo<double>(str.Value);
+                    decimal dec = ValueConverter.ConvertTo<decimal>(str.Value);
+                    // TODO: make more efficient, current solution allocates memory
```

Comment:
I will be glad if someone suggests a more efficient way.

Reply:
happy to go with this for now